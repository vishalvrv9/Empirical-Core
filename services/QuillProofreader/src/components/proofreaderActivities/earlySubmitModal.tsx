import * as React from 'react'

// const EarlySubmitModal: React.SFC<{closeModal: ((event: MouseEvent) => void), requiredEditCount:number}> = (props) => {
const EarlySubmitModal: React.SFC<{closeModal: any, requiredEditCount: number}> = (props) => {
 return <div className="early-submit-modal-container">
    <div className="early-submit-modal-background" />
     <div className="early-submit-modal">
       <div className="top-section">
          <h1>Keep Trying!</h1>
          <p>You must make at least {props.requiredEditCount} edits.</p>
          <button onClick={props.closeModal}>Find Edits</button>
        </div>
        </div>
     </div>
}

export default EarlySubmitModal
