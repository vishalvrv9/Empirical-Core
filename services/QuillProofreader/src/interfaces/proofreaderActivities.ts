export interface ProofreaderActivities {
  [key: string]: ProofreaderActivity;
}

export interface ProofreaderActivity {
  description: string;
  title: string;
  passage: string;
  standard: Standard;
  standardLevel: StandardLevel;
  topicCategory: TopicCategory;
  underlineErrorsInProofreader: boolean;
  flag?: string;
}

export interface Standard {
  name: string;
  uid: string;
}

export interface StandardLevel {
  name: string;
  uid: string;
}

export interface TopicCategory {
  name: string;
  uid: string;
}

export interface ConceptResultObject {
  metadata: ConceptResultMetadata,
  concept_uid: string,
  question_type: "passage-proofreader"
}

export interface ConceptResultMetadata {
  answer: string,
  correct: 0|1,
  instructions: string,
  prompt: string,
  questionNumber: number,
  unchanged: boolean,
}

export interface Concepts {
  [key: string]: Concept
}

export interface Concept {
  quantity: number;
}
