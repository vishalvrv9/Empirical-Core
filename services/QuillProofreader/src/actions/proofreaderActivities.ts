import rootRef from '../firebase';
import * as _ from 'lodash'
import { push } from 'react-router-redux';
import { ActionTypes } from './actionTypes'
const activitiesRef = rootRef.child('passageProofreadings')
import { ProofreaderActivities } from '../interfaces/proofreaderActivities'

export const startListeningToActivities = () => {
  return (dispatch: Function) => {
    activitiesRef.on('value', (snapshot: any) => {
      const activities: ProofreaderActivities[] = snapshot.val()
      if (activities) {
        dispatch({ type: ActionTypes.RECEIVE_PROOFREADER_ACTIVITIES_DATA, data: activities, });
      } else {
        dispatch({ type: ActionTypes.NO_PROOFREADER_ACTIVITIES_FOUND })
      }
    });

  }
}

export const getActivity = (activityUID: string) => {
  return (dispatch: Function) => {
    activitiesRef.child(activityUID).once('value', (snapshot: any) => {
      const activity: ProofreaderActivities = snapshot.val()
      if (activity) {
        dispatch({ type: ActionTypes.RECEIVE_PROOFREADER_ACTIVITY_DATA, data: activity, });
      } else {
        dispatch({ type: ActionTypes.NO_PROOFREADER_ACTIVITY_FOUND })
      }
    });

  }
}

export const toggleNewLessonModal = () => {
  return { type: ActionTypes.TOGGLE_NEW_LESSON_MODAL, };
}

export const submitNewLesson = (content: ProofreaderActivity) => {
  const cleanedContent = _.pickBy(content)
  return (dispatch: Function) => {
    dispatch({ type: ActionTypes.AWAIT_NEW_LESSON_RESPONSE, });
    const newRef = activitiesRef.push(cleanedContent, (error: string) => {
      dispatch({ type: ActionTypes.RECEIVE_NEW_LESSON_RESPONSE, });
      if (error) {
        dispatch({ type: ActionTypes.DISPLAY_ERROR, error: `Submission failed! ${error}`, });
      } else {
        dispatch({ type: ActionTypes.DISPLAY_MESSAGE, message: 'Submission successfully saved!', });
        const action = push(`/admin/lessons/${newRef.key}`);
        dispatch(action);
      }
    });
  };
}

export const startLessonEdit = (cid: string) => {
  return { type: ActionTypes.START_LESSON_EDIT, cid, };
}

export const cancelLessonEdit = (cid: string) => {
  return { type: ActionTypes.FINISH_LESSON_EDIT, cid, };
}

export const submitLessonEdit = (cid: string, content: GrammarActivity) => {
  return (dispatch: Function) => {
    dispatch({ type: ActionTypes.SUBMIT_LESSON_EDIT, cid, });
    const cleanedContent = _.pickBy(content)
    activitiesRef.child(cid).set(cleanedContent, (error: string) => {
      dispatch({ type: ActionTypes.FINISH_LESSON_EDIT, cid, });
      if (error) {
        dispatch({ type: ActionTypes.DISPLAY_ERROR, error: `Update failed! ${error}`, });
      } else {
        dispatch({ type: ActionTypes.DISPLAY_MESSAGE, message: 'Update successfully saved!', });
      }
    });
  };
}

export const deleteLesson = (cid: string) => {
  return (dispatch: Function) => {
    dispatch({ type: ActionTypes.SUBMIT_LESSON_EDIT, cid, });
    activitiesRef.child(cid).remove((error: string) => {
      dispatch({ type: ActionTypes.FINISH_LESSON_EDIT, cid, });
      if (error) {
        dispatch({ type: ActionTypes.DISPLAY_ERROR, error: `Deletion failed! ${error}`, });
      } else {
        dispatch({ type: ActionTypes.DISPLAY_MESSAGE, message: 'Lesson successfully deleted!', });
        const action = push(`/admin/lessons`)
        dispatch(action)
      }
    });
  };
}
