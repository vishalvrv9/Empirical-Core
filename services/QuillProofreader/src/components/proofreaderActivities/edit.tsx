import * as React from 'react'

import { Concept } from '../../interfaces/concepts'
const notNecessaryIconSrc = `${process.env.QUILL_CDN_URL}/images/icons/not_necessary_icon.png`
const incorrectIconSrc = `${process.env.QUILL_CDN_URL}/images/icons/incorrect_icon.png`
const correctIconSrc = `${process.env.QUILL_CDN_URL}/images/icons/correct_icon.png`

interface EditProps {
  displayText: string;
  state: string;
  concept: Concept|undefined;
  activeIndex: number;
  index: number;
  numberOfEdits: number;
  next: any;
  incorrectText: string|null;
  id: string;
}

export default class Edit extends React.Component<EditProps, any> {
  constructor(props: EditProps) {
    super(props)

    this.renderTooltip = this.renderTooltip.bind(this)
    this.renderGrammaticalConcept = this.renderGrammaticalConcept.bind(this)
    this.renderCorrectAnswers = this.renderCorrectAnswers.bind(this)
  }

  renderGrammaticalConcept() {
    const { concept } = this.props
    if (concept) {
      return <div>
        <p className="label">Grammatical Concept:</p>
        <p>{concept.name}</p>
      </div>
    }
  }

  renderCorrectAnswers() {
    const { displayText } = this.props
    if (displayText) {
      const correctAnswerArray = displayText ? displayText.split('~') : []
      let correctAnswers
      let correctAnswerHTML
      let labelText
      if (correctAnswerArray.length > 1) {
        correctAnswers = correctAnswerArray.map(ca => <li>{ca}</li>)
        correctAnswerHTML = <ul>{correctAnswers}</ul>
        labelText = 'Correct Edits:'
      } else {
        correctAnswers = correctAnswerArray[0]
        correctAnswerHTML = <p>{correctAnswers}</p>
        labelText = 'Correct Edit:'
      }
      return <div>
        <p className="label">{labelText}</p>
        {correctAnswerHTML}
      </div>
    } else {
      return <span/>
    }
  }

  renderTooltip() {
    const { activeIndex, index, state, numberOfEdits, next, id } = this.props
    const visible = activeIndex === index ? 'visible' : 'invisible'
    const element = document.getElementById(id)
    const offset = element && ((window.innerWidth - element.offsetLeft) < 350) ? 'offset' : ''
    const correctAnswers = this.props.displayText ? this.props.displayText.split('~') : ''
    let src, headerText
    switch (state) {
      case 'correct':
        src = correctIconSrc;
        headerText = 'Correct'
        break
      case 'incorrect':
        src = incorrectIconSrc;
        headerText = 'Incorrect'
        break
      case 'unnecessary':
        src = notNecessaryIconSrc
        headerText = 'Not Necessary'
        break
    }
    return <div className={`edit-tooltip ${visible} ${offset}`}>
      <div className="top-section">
        <div className="header">
          <img src={src}/>
          <h1>{headerText}</h1>
        </div>
        <p>Edit {index + 1} of {numberOfEdits}</p>
      </div>
      <div className="middle-section">
        {this.renderCorrectAnswers()}
        {this.renderGrammaticalConcept()}
      </div>
      <div className="button-section">
        <button onClick={next}>Next Edit âžž</button>
      </div>
    </div>
  }

  render() {
    const { activeIndex, index, state } = this.props
    const className = activeIndex === index ? 'active' : ''
    const tooltip = this.renderTooltip()
    return <div className={`edit ${className} ${state}`} id={this.props.id}>
      <strong>{this.props.incorrectText || this.props.displayText}</strong>
      {tooltip}
    </div>
  }
}
