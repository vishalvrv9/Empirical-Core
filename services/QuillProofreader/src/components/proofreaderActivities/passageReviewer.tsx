import * as React from 'react';

import { Concept } from '../../interfaces/concepts'
import Edit from './edit'

// import * as jsdiff from 'diff'
// import * as _ from 'underscore'

interface PassageReviewerProps {
  text: string;
  concepts: Concept[];
  finishReview: Function;
}

interface PassageReviewerState {
  activeIndex: number;
  numberOfEdits: number;
}

export default class PassageReviewer extends React.Component<PassageReviewerProps, PassageReviewerState> {
  constructor(props: PassageReviewerProps) {
    super(props)

    const matches = props.text ? props.text.match(/{\+([^-]+)-([^|]*)\|([^}]*)}/g) : []
    const numberOfEdits = matches ? matches.length : 0

    this.state = {
      activeIndex: 0,
      numberOfEdits
    }

    this.next = this.next.bind(this)
    this.renderFormattedText = this.renderFormattedText.bind(this)
  }

  componentDidMount() {
    window.scrollTo(0, 0)
  }

  next() {
    const { activeIndex, numberOfEdits } = this.state
    if (activeIndex + 1 === numberOfEdits) {
      this.props.finishReview()
    } else {
      this.setState({
        activeIndex: activeIndex + 1
      })
      const el = document.getElementById(String(activeIndex + 1))
      if (el) {
        el.scrollIntoView({ behavior: 'smooth', block: 'center' })
      }
    }
  }

  renderFormattedText() {
    const { text, concepts } = this.props
    const paragraphs = text.split('</p><p>')
    const punctuationRegex = /^[.,:;]/
    const { activeIndex, numberOfEdits } = this.state
    let index = 0
    return paragraphs.map((paragraph: string) => {
      const parts: Array<string|JSX.Element> = paragraph.replace(/<p>|<\/p>/g, '').split(/{|}/g)
      for (let i = 0; i < parts.length; i ++) {
        if (typeof parts[i] === "string" && parts[i][0] === '+') {
          const plusMatch = parts[i].match(/\+([^-]+)-/m)
          const plus = plusMatch ? plusMatch[1] : ''
          const conceptUIDMatch = parts[i].match(/\|([^-]+)/m)
          const conceptUID = conceptUIDMatch ? conceptUIDMatch[1] : ''
          const negativeMatch = parts[i].match(/\-([^-]+)\|/m)
          const negative = negativeMatch ? negativeMatch[1] : null
          const concept = this.props.concepts.find(c => c.uid === conceptUID)
          const indexToPass = index
          let state = 'correct'
          if (conceptUID === 'unnecessary') {
            state = 'unnecessary'
          } else if (negative) {
            state = 'incorrect'
          }
          index++
          parts[i] = <Edit
            state={state}
            concept={concept}
            displayText={plus}
            incorrectText={negative}
            index={indexToPass}
            activeIndex={activeIndex}
            next={this.next}
            numberOfEdits={numberOfEdits}
            id={`${index}`}
          />
          if (punctuationRegex.test(parts[i + 1])) {
            parts[i + 1] = `${parts[i + 1]}`
          } else {
            parts[i + 1] = ` ${parts[i + 1]}`
          }
        }
      }
      return <p>{parts}</p>
    })
  }

  render() {
    if (this.props.text) {
      return <div className="reviewer" >
        {this.renderFormattedText()}
      </div>
    } else {
      return <p>No passage</p>
    }
  }

}
