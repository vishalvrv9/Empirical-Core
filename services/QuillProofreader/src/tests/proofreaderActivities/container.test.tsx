import * as React from "react";
import {shallow} from "enzyme";
import { PlayProofreaderContainer } from "../../components/proofreaderActivities/container";
import { ProofreaderActivityReducer } from './data'

process.env.EMPIRICAL_BASE_URL = 'https://staging.quill.org'
process.env.QUILL_CMS = 'https://cms.quill.org'

describe("<PlayProofreaderContainer />", () => {
  const wrapper = shallow(<PlayProofreaderContainer
    dispatch={() => {}}
    activityUID='KMyh3LulfVL0_KuPb8u'
    session={{passage: 'string'}}
    proofreaderActivities={ProofreaderActivityReducer}
   />)

    it("should render", () => {
      expect(wrapper).toMatchSnapshot();
    });

});
