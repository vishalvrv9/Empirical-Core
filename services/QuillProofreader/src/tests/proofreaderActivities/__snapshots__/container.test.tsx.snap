// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlayProofreaderContainer /> should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlayProofreaderContainer
    activityUID="KMyh3LulfVL0_KuPb8u"
    dispatch={[Function]}
    proofreaderActivities={
      Object {
        "currentActivity": Object {
          "description": "Correct the following nine errors.",
          "flag": "production",
          "passage": "Coney Island is both a neighborhood in {+Brooklyn, New York,-Brooklyn New York|1AJtvwkbBLHzYg2a4eRcBA} and an amusement park. {+There are-They’re|8NObTn-ErGu5tADF0LWxbQ} many tourist attractions at Coney Island like the Cyclone rollercoaster and the Wonder Wheel ferris wheel. Coney means rabbit. There is a theory that the island is named Coney because there was a large population of wild rabbits when the area was {+an-a|5s-r1281DV8EpHyc9qJfiw} English colony in the 1600s. The Cyclone roller coaster was built in 1927, and it is one of the few wooden roller coasters still in use in the United States. The Cyclones {+is-are|Tlhrx6Igxn6cR_SD1U5efA} also the name of the minor league baseball team that plays at the stadium in the neighborhood.  There is a  parachute jump from the 1939 {+World's-Worlds|nAcT-C3UfPFuhWcf0JJNMw} Fair still intact, and although it is no longer used, it is a landmark occasionally referred {+to-too|Oo9MdluHtJFdh5d1uL9qYQ} as the Effiel Tower of Brooklyn. The New York Aquarium is near Coney Island, just one subway stop away at a station {+called-call|IDyiKUM8LLD7Bz8ftg6p0A} West Eighth Street. The Coney Island stop is the last stop on the F line subway. <br/><br/>

Over the years, Coney Island has expanded to have a few parks with rides. There is a park called {+Astroland-astroland|YkA1YFe-dUvXqkEXxbNgQw} and another section called Luna Park that has rides for smaller kids. The boardwalk of Coney Island is thriving with food stands, including the original Nathan’s Hot Dog stand. Every year at Coney Island, Nathan’s hosts a hot dog eating contest. The winner of the contest usually  eats around 150 hot dogs in an {+extremely-extreme|Jm8fUXNgIR66RYWt51OVyQ} short period of time. Also annually at Coney Island is the Mermaid Parade, where people dress up as mermaids. The mayor of New York is always in attendance.<br/><br/>",
          "standard": Object {
            "name": "CCSS Grade 7 Summative Assessments",
            "uid": "pM3sNYWnNnWdR5YKu7lAHw",
          },
          "standardLevel": Object {
            "name": "7th Grade CCSS",
            "uid": "8g1_400wzXuijt4NB9uqtw",
          },
          "title": "Coney Island",
          "topicCategory": Object {
            "name": "Summative Assessments",
            "uid": "XSx6hcLxlSf1WKaJFWgsNA",
          },
          "underlineErrorsInProofreader": false,
        },
        "data": Object {
          "-KMyh3LulfVL0_KuPb8u": Object {
            "description": "Correct the following nine errors.",
            "flag": "production",
            "passage": "Coney Island is both a neighborhood in {+Brooklyn, New York,-Brooklyn New York|1AJtvwkbBLHzYg2a4eRcBA} and an amusement park. {+There are-They’re|8NObTn-ErGu5tADF0LWxbQ} many tourist attractions at Coney Island like the Cyclone rollercoaster and the Wonder Wheel ferris wheel. Coney means rabbit. There is a theory that the island is named Coney because there was a large population of wild rabbits when the area was {+an-a|5s-r1281DV8EpHyc9qJfiw} English colony in the 1600s. The Cyclone roller coaster was built in 1927, and it is one of the few wooden roller coasters still in use in the United States. The Cyclones {+is-are|Tlhrx6Igxn6cR_SD1U5efA} also the name of the minor league baseball team that plays at the stadium in the neighborhood.  There is a  parachute jump from the 1939 {+World's-Worlds|nAcT-C3UfPFuhWcf0JJNMw} Fair still intact, and although it is no longer used, it is a landmark occasionally referred {+to-too|Oo9MdluHtJFdh5d1uL9qYQ} as the Effiel Tower of Brooklyn. The New York Aquarium is near Coney Island, just one subway stop away at a station {+called-call|IDyiKUM8LLD7Bz8ftg6p0A} West Eighth Street. The Coney Island stop is the last stop on the F line subway. <br/><br/>

Over the years, Coney Island has expanded to have a few parks with rides. There is a park called {+Astroland-astroland|YkA1YFe-dUvXqkEXxbNgQw} and another section called Luna Park that has rides for smaller kids. The boardwalk of Coney Island is thriving with food stands, including the original Nathan’s Hot Dog stand. Every year at Coney Island, Nathan’s hosts a hot dog eating contest. The winner of the contest usually  eats around 150 hot dogs in an {+extremely-extreme|Jm8fUXNgIR66RYWt51OVyQ} short period of time. Also annually at Coney Island is the Mermaid Parade, where people dress up as mermaids. The mayor of New York is always in attendance.<br/><br/>",
            "standard": Object {
              "name": "CCSS Grade 7 Summative Assessments",
              "uid": "pM3sNYWnNnWdR5YKu7lAHw",
            },
            "standardLevel": Object {
              "name": "7th Grade CCSS",
              "uid": "8g1_400wzXuijt4NB9uqtw",
            },
            "title": "Coney Island",
            "topicCategory": Object {
              "name": "Summative Assessments",
              "uid": "XSx6hcLxlSf1WKaJFWgsNA",
            },
            "underlineErrorsInProofreader": false,
          },
        },
        "hasreceiveddata": true,
        "states": Object {},
      }
    }
    session={
      Object {
        "passage": "string",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="header-section"
        >
          <div
            className="inner-header"
          >
            <h1>
              Coney Island
            </h1>
            <div
              className="instructions"
            >
              <div>
                <img
                  src="undefined/images/icons/question_icon.svg"
                />
                <p
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "Correct the following nine errors.",
                    }
                  }
                />
              </div>
              <div
                className="edits-made"
              >
                <p>
                  Edits Made: 
                  0
                   of 
                  0
                </p>
                <div
                  className="progress-bar-indication"
                >
                  <span
                    className="meter"
                    style={
                      Object {
                        "width": "0%",
                      }
                    }
                  />
                </div>
              </div>
            </div>
          </div>
        </div>,
        undefined,
        undefined,
        <div
          className="passage "
        />,
        <div
          className="bottom-section"
        >
          <button
            onClick={[Function]}
          >
            Check Work
          </button>
        </div>,
      ],
      "className": "passage-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="inner-header"
          >
            <h1>
              Coney Island
            </h1>
            <div
              className="instructions"
            >
              <div>
                <img
                  src="undefined/images/icons/question_icon.svg"
                />
                <p
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "Correct the following nine errors.",
                    }
                  }
                />
              </div>
              <div
                className="edits-made"
              >
                <p>
                  Edits Made: 
                  0
                   of 
                  0
                </p>
                <div
                  className="progress-bar-indication"
                >
                  <span
                    className="meter"
                    style={
                      Object {
                        "width": "0%",
                      }
                    }
                  />
                </div>
              </div>
            </div>
          </div>,
          "className": "header-section",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                Coney Island
              </h1>,
              <div
                className="instructions"
              >
                <div>
                  <img
                    src="undefined/images/icons/question_icon.svg"
                  />
                  <p
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "Correct the following nine errors.",
                      }
                    }
                  />
                </div>
                <div
                  className="edits-made"
                >
                  <p>
                    Edits Made: 
                    0
                     of 
                    0
                  </p>
                  <div
                    className="progress-bar-indication"
                  >
                    <span
                      className="meter"
                      style={
                        Object {
                          "width": "0%",
                        }
                      }
                    />
                  </div>
                </div>
              </div>,
            ],
            "className": "inner-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Coney Island",
              },
              "ref": null,
              "rendered": "Coney Island",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div>
                    <img
                      src="undefined/images/icons/question_icon.svg"
                    />
                    <p
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "Correct the following nine errors.",
                        }
                      }
                    />
                  </div>,
                  <div
                    className="edits-made"
                  >
                    <p>
                      Edits Made: 
                      0
                       of 
                      0
                    </p>
                    <div
                      className="progress-bar-indication"
                    >
                      <span
                        className="meter"
                        style={
                          Object {
                            "width": "0%",
                          }
                        }
                      />
                    </div>
                  </div>,
                ],
                "className": "instructions",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <img
                        src="undefined/images/icons/question_icon.svg"
                      />,
                      <p
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "Correct the following nine errors.",
                          }
                        }
                      />,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "src": "undefined/images/icons/question_icon.svg",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "img",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "dangerouslySetInnerHTML": Object {
                          "__html": "Correct the following nine errors.",
                        },
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "p",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <p>
                        Edits Made: 
                        0
                         of 
                        0
                      </p>,
                      <div
                        className="progress-bar-indication"
                      >
                        <span
                          className="meter"
                          style={
                            Object {
                              "width": "0%",
                            }
                          }
                        />
                      </div>,
                    ],
                    "className": "edits-made",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          "Edits Made: ",
                          0,
                          " of ",
                          0,
                        ],
                      },
                      "ref": null,
                      "rendered": Array [
                        "Edits Made: ",
                        0,
                        " of ",
                        0,
                      ],
                      "type": "p",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <span
                          className="meter"
                          style={
                            Object {
                              "width": "0%",
                            }
                          }
                        />,
                        "className": "progress-bar-indication",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "className": "meter",
                          "style": Object {
                            "width": "0%",
                          },
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "span",
                      },
                      "type": "div",
                    },
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      undefined,
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "passage ",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            onClick={[Function]}
          >
            Check Work
          </button>,
          "className": "bottom-section",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Check Work",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Check Work",
          "type": "button",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="header-section"
          >
            <div
              className="inner-header"
            >
              <h1>
                Coney Island
              </h1>
              <div
                className="instructions"
              >
                <div>
                  <img
                    src="undefined/images/icons/question_icon.svg"
                  />
                  <p
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "Correct the following nine errors.",
                      }
                    }
                  />
                </div>
                <div
                  className="edits-made"
                >
                  <p>
                    Edits Made: 
                    0
                     of 
                    0
                  </p>
                  <div
                    className="progress-bar-indication"
                  >
                    <span
                      className="meter"
                      style={
                        Object {
                          "width": "0%",
                        }
                      }
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>,
          undefined,
          undefined,
          <div
            className="passage "
          />,
          <div
            className="bottom-section"
          >
            <button
              onClick={[Function]}
            >
              Check Work
            </button>
          </div>,
        ],
        "className": "passage-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="inner-header"
            >
              <h1>
                Coney Island
              </h1>
              <div
                className="instructions"
              >
                <div>
                  <img
                    src="undefined/images/icons/question_icon.svg"
                  />
                  <p
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "Correct the following nine errors.",
                      }
                    }
                  />
                </div>
                <div
                  className="edits-made"
                >
                  <p>
                    Edits Made: 
                    0
                     of 
                    0
                  </p>
                  <div
                    className="progress-bar-indication"
                  >
                    <span
                      className="meter"
                      style={
                        Object {
                          "width": "0%",
                        }
                      }
                    />
                  </div>
                </div>
              </div>
            </div>,
            "className": "header-section",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1>
                  Coney Island
                </h1>,
                <div
                  className="instructions"
                >
                  <div>
                    <img
                      src="undefined/images/icons/question_icon.svg"
                    />
                    <p
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "Correct the following nine errors.",
                        }
                      }
                    />
                  </div>
                  <div
                    className="edits-made"
                  >
                    <p>
                      Edits Made: 
                      0
                       of 
                      0
                    </p>
                    <div
                      className="progress-bar-indication"
                    >
                      <span
                        className="meter"
                        style={
                          Object {
                            "width": "0%",
                          }
                        }
                      />
                    </div>
                  </div>
                </div>,
              ],
              "className": "inner-header",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Coney Island",
                },
                "ref": null,
                "rendered": "Coney Island",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div>
                      <img
                        src="undefined/images/icons/question_icon.svg"
                      />
                      <p
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "Correct the following nine errors.",
                          }
                        }
                      />
                    </div>,
                    <div
                      className="edits-made"
                    >
                      <p>
                        Edits Made: 
                        0
                         of 
                        0
                      </p>
                      <div
                        className="progress-bar-indication"
                      >
                        <span
                          className="meter"
                          style={
                            Object {
                              "width": "0%",
                            }
                          }
                        />
                      </div>
                    </div>,
                  ],
                  "className": "instructions",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <img
                          src="undefined/images/icons/question_icon.svg"
                        />,
                        <p
                          dangerouslySetInnerHTML={
                            Object {
                              "__html": "Correct the following nine errors.",
                            }
                          }
                        />,
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "src": "undefined/images/icons/question_icon.svg",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "img",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "dangerouslySetInnerHTML": Object {
                            "__html": "Correct the following nine errors.",
                          },
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "p",
                      },
                    ],
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <p>
                          Edits Made: 
                          0
                           of 
                          0
                        </p>,
                        <div
                          className="progress-bar-indication"
                        >
                          <span
                            className="meter"
                            style={
                              Object {
                                "width": "0%",
                              }
                            }
                          />
                        </div>,
                      ],
                      "className": "edits-made",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": Array [
                            "Edits Made: ",
                            0,
                            " of ",
                            0,
                          ],
                        },
                        "ref": null,
                        "rendered": Array [
                          "Edits Made: ",
                          0,
                          " of ",
                          0,
                        ],
                        "type": "p",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": <span
                            className="meter"
                            style={
                              Object {
                                "width": "0%",
                              }
                            }
                          />,
                          "className": "progress-bar-indication",
                        },
                        "ref": null,
                        "rendered": Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "className": "meter",
                            "style": Object {
                              "width": "0%",
                            },
                          },
                          "ref": null,
                          "rendered": null,
                          "type": "span",
                        },
                        "type": "div",
                      },
                    ],
                    "type": "div",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        undefined,
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "passage ",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              onClick={[Function]}
            >
              Check Work
            </button>,
            "className": "bottom-section",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Check Work",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Check Work",
            "type": "button",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
